codebundles/istio-sidecar-check/runbook.robot

```diff
@@ -0,0 +1,129 @@
+*** Settings ***
+Documentation       Checks for Istio sidecar injection across deployments in all namespaces.
+Metadata           Author    Helpful Assistant
+Metadata           Display Name    Istio Sidecar Injection Check
+Metadata           Supports    Kubernetes,AKS,EKS,GKE,OpenShift
+
+Library             BuiltIn
+Library             RW.Core
+Library             RW.CLI
+Library             RW.platform
+Library             RW.K8sHelper
+Library             OperatingSystem
+Library             String
+
+Suite Setup         Suite Initialization
+
+*** Tasks ***
+
+Check Deployments For Istio Sidecar Injection
+    [Documentation]    Checks all deployments in specified namespaces for Istio sidecar injection status
+    [Tags]    
+    ...    istio
+    ...    sidecar
+    ...    injection
+    ...    deployment
+    ${results}=    RW.CLI.Run Bash File
+    ...    bash_file=check_istio_injection.sh
+    ...    env=${env}
+    ...    secret_file__kubeconfig=${kubeconfig}
+    ...    include_in_history=true
+    
+    # Save command history for report
+    ${history}=    RW.CLI.Pop Shell History
+    ${commands_used}=    Set Variable    ${history.strip()}
+    Create File    command_history.txt    Commands Used:\n${commands_used}
+
+    # Extract issues and report from stdout
+    ${stdout_lines}=    Split To Lines    ${results.stdout}
```

Comment:
Consider refactoring this block and hide the processing in fewer steps -  it's very challenging to read. 

One  common pattern is to, in your script, write out the necessary report and issue data to two separate files if you like. Simply add issues to a dedicated json file and then read that in for processing of the issues. 

For example, this script writes the issues to a file: https://github.com/runwhen-contrib/rw-cli-codecollection/blob/main/codebundles/k8s-pvc-healthcheck/pvc_utilization_check.sh

This code reads it in and processes the issues
https://github.com/runwhen-contrib/rw-cli-codecollection/blob/cb6fe6c4e74767bb933318326f2a93b6e2ee33fa/codebundles/k8s-pvc-healthcheck/runbook.robot#L153